src/irc/core/irc-servers.c

```diff
@@ -313,9 +313,9 @@ static void server_init_1(IRC_SERVER_REC *server)
 	if (!conn->no_cap) {
 		signal_emit("server waiting cap ls", 2, server, CAP_LS_VERSION);
 		irc_send_cmd_now(server, "CAP LS " CAP_LS_VERSION);
-		/* to detect non-CAP servers, send this bogus join */
-		/* the : will make INSPIRCD respond with 451 instead of 461, too */
-		irc_send_cmd_now(server, "JOIN :");
+
+		if (!server->cap_supported)
```

Comment:
`server->cap_supported` will always be `NULL` at this point, this is entirely wrong.

Reply:
Tried to fix it. Sorry if it's still "not quite right"; on my system and setup it "just works".

https://github.com/MarioMario456/irssi/commit/d7e1bdba8202a0f5f5259152a05c1ca61042cb63